src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -35,6 +35,9 @@ public User register(String email, String password) throws RegistrationException
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("Email Already Exists");
+        }
```

Comment:
You could do this check much earlier to avoid redundant user creation in case the email already exists in DB.